nsqd/nsqd.go

```diff
@@ -370,6 +379,11 @@ func (n *NSQD) PersistMetadata() error {
 		return err
 	}
 
+	err = os.Remove(oldFileName)
```

Comment:
The concern is that some messages will be processed before the version rollback, so the old metadata file will have stale info, be missing new messages, have references to queue files that don't exist anymore...